### PR TITLE
Add ability to upload scripts used with exec plugin

### PR DESCRIPTION
**Pull request by matejzero**
I'm no expert in puppet and I don't know enough to write tests, but I'm using this hack to be able to upload shell scripts together with exec plugin. It's based on python plugin.

Example:
```puppet
  class { 'collectd::plugin::exec':
    commands => {
      'sockstat'            => {
        user          => 'collector',
        group         => 'collector',
        script_source => 'puppet:///modules/collectd/sockstat.sh',
        exec          => ['/etc/collectd.d/sockstat.sh'],
      },
      'network_connections' => {
        user          => 'collector',
        group         => 'collector',
        script_source => 'puppet:///modules/collectd/network_connections.sh',
        exec          => ['/etc/collectd.d/network_connections.sh'],
      },
      'smartmon'            => {
        user          => 'collector',
        group         => 'collector',
        script_source => 'puppet:///modules/collectd/smartmon.sh',
        exec          => ['/etc/collectd.d/smartmon.sh'],
        extra_param   => ['sda','sda'],
      },
    }
  }
```

### CODE REVIEW

File: templates/plugin/exec/cmd.conf.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <%- unless @exec.empty? -%>
-  Exec "<%= @user %>:<%= @group %>" "<%= @exec.join('" "') %>"
+  Exec "<%= @user %>:<%= @group %>" "<%= @exec.join('" "') %>" <% if @extra_param %><% @extra_param.each do |extra| -%>"<%= extra %>" <% end -%><% end -%>
```
Comment (igalic) [inline templates/plugin/exec/cmd.conf.erb]: what you're trying to say here is easier expressed in a join:
```erb
<% if @extra_param %><%= '"' + @extra_param.join ('" "') + '"' %><% end %>
```

File: manifests/plugin/exec/cmd.pp [hunk 2]
```diff
@@ -3,7 +3,10 @@
   $group,
   $exec              = [],
   $notification_exec = [],
-  $ensure = present,
+  $script_source     = undef,
```
Comment (igalic) [inline manifests/plugin/exec/cmd.pp]: i'd also offer `$script_content`

**Comment by igalic** [comment]
@matejzero thank you very much for this patch!

aside from the comments above, here are some general ones:

* this needs an update of the [README to document the new feature](https://github.com/puppet-community/puppet-collectd#class-collectdpluginexec)
* it would be also great to have [spec tests](https://github.com/puppet-community/puppet-collectd/blob/master/spec/defines/collectd_plugin_exec_cmd_spec.rb)
* you'll probably have an easier time modifying patches when you follow the [github flow workflow](https://guides.github.com/introduction/flow/index.html)

if you need help, you can find me or other helpful people in #puppet-community on freenode

**Comment by blkperl** [comment]
Closing due to lack of activity. Please reopen when the feedback has been addressed

**Pull request closed without merging**